census/ui_app/QuestionForm.jsx

```diff
@@ -186,20 +233,66 @@ const QuestionForm = React.createClass({
       if (this.getSchemaForId(q.id) === undefined) {
         console.warn('No schema defined for Question with id: ' + q.id);
       }
-      return (
-        <QuestionField ref={q.id}
-                       key={q.id}
-                       id={q.id}
-                       visibleProps={this.getVisiblePropsForId(q.id)}
-                       value={q.value}
-                       onChange={this.onFieldChange}
-                       label={this.getLabelForId(q.id)}
-                       position={this.getPositionForId(q.id)}
-                       instructions={this.getValueForId(q.id, 'description')}
-                       placeholder={this.getValueForId(q.id, 'placeholder')}>
-          {this.getValueForId(q.id, 'text')}
-        </QuestionField>
-      );
+      let questionNode = '';
+      switch (this.getValueForId(q.id, 'type')) {
```

Comment:
I see the component properties are the same in the following chunk of code. So, instead of violating DRY rules, I'd do something like:

```javascript
const types = {
  'yesno': 'YesNo',
  'text': 'Text'
};

// Here i'm not sure how to reference to the `type` in the `switch` block
<QuestionField{types[type]}
  // All the values between the tags, that are repeated otherwise
>
</QuestionField{types[type]}
```

Worst case scenario, you could leave it as it is, but grab all the properties together in a text variable and pass it along. Not sure if it works without any React-ish specific binding though, so I'd stick to manipulating the tag names instead.